services/device-registry-file/src/test/java/org/eclipse/hono/deviceregistry/file/FileBasedDeviceManagementSearchDevicesTest.java

```diff
@@ -0,0 +1,146 @@
+/*****************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.deviceregistry.file;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.net.HttpURLConnection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.hono.deviceregistry.util.DeviceRegistryUtils;
+import org.eclipse.hono.service.management.device.AbstractDeviceManagementSearchDevicesTest;
+import org.eclipse.hono.service.management.device.Device;
+import org.eclipse.hono.service.management.device.DeviceManagementService;
+import org.eclipse.hono.service.management.device.Filter;
+import org.eclipse.hono.service.management.device.SearchDevicesResult;
+import org.eclipse.hono.service.management.device.Sort;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.opentracing.Span;
+import io.opentracing.noop.NoopSpan;
+import io.vertx.core.Vertx;
+import io.vertx.core.file.FileSystem;
+import io.vertx.junit5.Timeout;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+
+/**
+ * Tests for {@link FileBasedRegistrationService#searchDevices(String, int, int, List, List, Span)}.
+ */
+@ExtendWith(VertxExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@Timeout(value = 5, timeUnit = TimeUnit.SECONDS)
+public final class FileBasedDeviceManagementSearchDevicesTest implements AbstractDeviceManagementSearchDevicesTest {
+
+    private static final String FILE_NAME = "/device-identities.json";
+
+    private static final Logger LOG = LoggerFactory.getLogger(FileBasedDeviceManagementSearchDevicesTest.class);
+    private Vertx vertx;
+    private FileBasedRegistrationService registrationService;
+    private FileBasedRegistrationConfigProperties registrationConfig;
+    private FileSystem fileSystem;
+
+    /**
+     * Sets up static fixture.
+     *
+     * @param testContext The test context to use for running asynchronous tests.
+     */
+    @BeforeAll
+    public void setup(final VertxTestContext testContext) {
```

Comment:
`testContext` is not being set to complete properly. It is one of the reasons for the tests failure.
This can be removed as it is not being used insde the `setup(...)` method.

Reply:
Thanks for figuring it out !